src/sentry/models/groupresolution.py

```diff
@@ -69,6 +73,35 @@ def has_resolution(cls, group, release):
             return True
 
         if res_type in (None, cls.Type.in_next_release):
+            # If current_release_version was set, it means we are probably using semver since
+            # current_release_version is only set after checking that the project is following
+            # semver versioning scheme
+            if current_release_version:
+                follows_semver = follows_semver_versioning_scheme(
+                    project_id=group.project.id,
+                    org_id=group.organization.id,
+                    release_version=release.version,
+                )
+                # However even though we are almost sure that we are following semver, we need to
+                # check once more to ensure that both the release version passed follows semver
+                # and that the user did not stop using semver
+                if follows_semver:
+                    try:
+                        # If current_release_version == release.version => 0
+                        # If current_release_version < release.version => -1
+                        # If current_release_version > release.version => 1
+                        current_release_raw = parse_release(current_release_version).get(
+                            "version_raw"
+                        )
+                        release_raw = parse_release(release.version).get("version_raw")
+                        return compare_version_relay(current_release_raw, release_raw) == -1
```

Comment:
What happens if the releases are from different packages? Or is this just really a concern?

Reply:
Good point, I spoke to Armin about this and he told me even if the packages are different we still discard the package and compare the versions because the expectation here is that if you are within the same project, then this resolved behaviour should be done across packages 

for example lets say you have russia@1.0.0, austria@1.0.0, usa@1.0.0, and they are all within the same project, then if you resolve in next semver in one, it should happen in all 

If on the other hand, you want to have different package release branches then you need to separate with projects